### PR TITLE
Remove NioNotEnabledBootstrapCheck bootstrap check

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This is related to #27260. This commit removes the bootstrap check that
prevents nio from being enabled.

### CODE REVIEW

File: plugins/transport-nio/src/main/java/org/elasticsearch/transport/nio/NioNotEnabledBootstrapCheck.java [hunk 1]
```diff
@@ -21,12 +21,23 @@
 
 import org.elasticsearch.bootstrap.BootstrapCheck;
 import org.elasticsearch.bootstrap.BootstrapContext;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.util.concurrent.EsExecutors;
```
Comment (danielmitterdorfer) [inline plugins/transport-nio/src/main/java/org/elasticsearch/transport/nio/NioNotEnabledBootstrapCheck.java]: Nit: I think this import should not be needed?

File: plugins/transport-nio/src/main/java/org/elasticsearch/transport/nio/NioNotEnabledBootstrapCheck.java [hunk 2]
```diff
@@ -21,12 +21,23 @@
...
+
+import static org.elasticsearch.common.settings.Setting.boolSetting;
 
 public class NioNotEnabledBootstrapCheck implements BootstrapCheck {
 
+    public static final Setting<Boolean> OVERRIDE_BOOTSTRAP =
```
Comment (danielmitterdorfer) [inline plugins/transport-nio/src/main/java/org/elasticsearch/transport/nio/NioNotEnabledBootstrapCheck.java]: Tbh, I was not sure whether a system property or a setting would be a better choice but using a setting for this is consistent with what we did previously (see e.g. #20511).

File: plugins/transport-nio/src/main/java/org/elasticsearch/transport/nio/NioNotEnabledBootstrapCheck.java [hunk 3]
```diff
@@ -21,12 +21,23 @@
...
+        boolSetting("transport.nio.override_nio_bootstrap_check", false, Setting.Property.NodeScope);
```
Comment (danielmitterdorfer) [inline plugins/transport-nio/src/main/java/org/elasticsearch/transport/nio/NioNotEnabledBootstrapCheck.java]: Alternatively `bootstrap.ignore_transport_nio_bootstrap_checks`?

**Comment by jasontedor** [comment]
Is this bootstrap check still needed? Can we remove it and only add it back if we decide NIO is not ready for production usage before 7.0.0?

**Comment by tbrooks8** [comment]
@jasontedor I think that is a question for @s1monw. He asked me to add the bootstrap check in one of my past PRs.

**Comment by jasontedor** [comment]
Sure thing let us wait for @s1monw, it's impotent though once there is a flag to skip it.

**Comment by s1monw** [comment]
> Is this bootstrap check still needed? Can we remove it and only add it back if we decide NIO is not ready for production usage before 7.0.0?

++ just remove it for now

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**